cg/apps/downsample/downsample.py

```diff
@@ -0,0 +1,169 @@
+            f"Application tag set to: {downsampled_sample.application_version.application.tag}"
+            f"Customer set to: {downsampled_sample.customer}"
+        )
+        if not self.dry_run:
+            self.status_db.session.add(downsampled_sample)
+            self.status_db.session.commit()
+            LOG.info(f"Added {downsampled_sample.name} to StatusDB.")
+            return downsampled_sample
+        return downsampled_sample
+
+    def add_downsampled_case_to_statusdb(self) -> Family | None:
+        """
+        Add a down sampled case entry to StatusDB.
+        """
+        downsampled_case: Family = self.downsample_data.downsampled_case
+        if case_exists_in_statusdb(status_db=self.status_db, case_name=downsampled_case.name):
+            LOG.info(f"Case with name {downsampled_case.name} already exists in StatusDB.")
+            return
+        if not self.dry_run:
+            self.status_db.session.add(downsampled_case)
+            self.status_db.session.commit()
+            LOG.info(f"New down sampled case created: {downsampled_case.internal_id}")
+            return downsampled_case
+        return downsampled_case
+
+    def _link_downsampled_sample_to_case(self, sample: Sample, case: Family) -> None:
+        """Create a link between sample and case in statusDB."""
+        if self.dry_run:
+            LOG.info(
+                f"Would relate sample {sample} to case {case.internal_id} with name {case.name}"
+            )
+            return
+
+        sample_case_link: FamilySample = self.status_db.relate_sample(
+            family=case,
+            sample=sample,
+            status=self.downsample_data.sample_status(sample=sample),
+        )
+        self.status_db.session.add(sample_case_link)
+        self.status_db.session.commit()
+        LOG.info(f"Related sample {sample.internal_id} to {case.internal_id}")
+
+    def add_downsampled_sample_case_to_statusdb(self) -> None:
+        """
+        Add the downsampled sample and case to statusDB and generate the sample case link.
+        """
+        self.add_downsampled_sample_entry_to_statusdb()
+        self.add_downsampled_case_to_statusdb()
+        self._link_downsampled_sample_to_case(
+            sample=self.downsample_data.downsampled_sample,
+            case=self.downsample_data.downsampled_case,
+        )
+
+    def is_decompression_needed(self) -> bool:
+        """Check if decompression is needed for the specified case.
+        Decompression is needed if there are no files with fastq tag found for the sample in housekeeper.
+        """
+        LOG.debug("Checking if decompression is needed.")
+        is_decompression_needed: bool = False
+        if not self.housekeeper_api.get_files(
+            bundle=self.downsample_data.original_sample.internal_id,
+            tags=[SequencingFileTag.FASTQ],
+        ):
+            is_decompression_needed = True
+        return is_decompression_needed
```

Comment:
That is based on case, but I need it for a sample only.

Reply:
Oh right! Then the docstring should probably be updated. But there is still a point in getting the compression objects for the sample and returning in a similar manner as in that method, since I think the suggested approach might need to support ongoing decompressions. In an ideal world, maybe the is_spring_decompression_needed method should be refactored so that we can stay DRY, but that is probably out of scope and can be done at a later time. 

Return statement for the aforementioned decompression method:
```Python
return any(
            not self.crunchy_api.is_compression_pending(compression_object)
            and not compression_object.pair_exists()
            for compression_object in compression_objects
        )
```

Reply:
Yes this was the initial method I used, but here I ran into errors when t would want to decompress the whole case instead of just the sample.

Reply:
added more extensive checks